aequilibrae/transit/transit_graph_builder.py

```diff
@@ -1058,18 +1061,103 @@ def create_edges(self):
         self.edges.d_line_id = self.edges.d_line_id.fillna("").astype(str)
         self.edges.direction = self.edges.direction.astype("int8")
 
-    def create_line_geometry(self):
+    def create_line_geometry(self, method="direct"):
         """Create the LineString for each edge."""
-        lines = []
-        for row in self.edges.itertuples():
-            # row.a_node - 1 because the node_ids are the index + 1
-            line = (
-                shapely.from_wkb(self.vertices.at[row.a_node - 1, "geometry"]),
-                shapely.from_wkb(self.vertices.at[row.b_node - 1, "geometry"]),
+        assert method in ["direct", "connector project match"]
+
+        if method == "direct":
+            self.edges["geometry"] = [
+                shapely.LineString(
+                    (
+                        shapely.from_wkb(self.vertices.at[row.a_node - 1, "geometry"]),
+                        shapely.from_wkb(self.vertices.at[row.b_node - 1, "geometry"]),
+                    )
+                ).wkb
+                for row in self.edges.itertuples()
+            ]
+        elif method == "connector project match":
```

Comment:
probably makes sense to extract a method rather than having 100 lines in a else statement